lib/adapters/QUnitAdapter.js

```diff
@@ -0,0 +1,73 @@
+import EventEmitter from "../EventEmitter.js"
+import {Test, Suite} from "../Data.js"
+
+
+export default class QUnitAdapter extends EventEmitter {
+    constructor() {
+        super();
+        QUnit.done(this.onDone.bind(this));
+        QUnit.testDone(this.onTestDone.bind(this));
+        QUnit.moduleDone(this.onModuleDone.bind(this));
+        QUnit.log(this.onLog.bind(this));
+        QUnit.testStart(this.onTestStart.bind(this));
+
+        this.tests = {};
+        this.suites = [];
+    }
+
+    onTestStart() {
+        this.errors = [];
```

Comment:
I guess `runStart` is the more interesting event. The number of tests is obviously pretty important there. Not just for TAP, Karma also needs (at least, wants) that information.